### DISCUSSION TITLE
ARIA input + label relationships

### DISCUSSION THREAD

**Discussion by SeanFree** [quasarframework/quasar#15361] [General - Components / Directives / etc]
I'm new to using Quasar, so apologies if I just overlooked something, I searched for similar topics but couldn't find anything.

I'd like to be able to apply an `id` to Input components in order to associate labels and descriptors for 2 reasons:
- Support screen readers
- Focus the input when the label is clicked (default label -> input behavior)

Ex:
```
<label :for="inputId">{{ label }}</label>
<QInput :id="inputId" v-model="model" />
 ```
 
 Currently it appears that `id` is being dropped / ignored in fallthrough props on all Input types (that I've tried), so there doesn't seem to be a way to do this.

I tried nesting the Input inside the label but some inputs (e.g. QSlider) are difficult to style, seemingly unresponsive to any styles I tried. Also, doing this still doesn't enable the click-to-focus behavior for labels.

**Comment by pdanpdan** [comment]
For QInput and other QField derivates use the `for` prop

**Reply by SeanFree** [reply]
Ok I definitely overlooked QField, thanks for the reply!
